tests/js/server-test-caseholder.js

```diff
@@ -0,0 +1,713 @@
+};
+
+// Each test has a function that is called when its request is issued.
+gpii.express.couchuser.test.server.caseHolder.verifyLoggedIn = function (response, body) {
+    gpii.express.couchuser.test.server.caseHolder.isSaneResponse(response, body);
+
+    var data = typeof body === "string" ? JSON.parse(body) : body;
+    jqUnit.assertTrue("The response should be 'ok'.", data.ok);
+    jqUnit.assertNotNull("There should be a user returned.", data.user);
+    if (data.user) {
+        jqUnit.assertEquals("The current user should be returned.", "admin", data.user.name);
+    }
+};
+
+gpii.express.couchuser.test.server.caseHolder.verifyCurrentUserLoggedIn = function (response, body) {
+    gpii.express.couchuser.test.server.caseHolder.isSaneResponse(response, body);
+
+    var data = typeof body === "string" ? JSON.parse(body) : body;
+    jqUnit.assertTrue("The response should be 'ok'.", data.ok);
+    jqUnit.assertNotNull("There should be a user returned.", data.user);
+    if (data.user) {
+        jqUnit.assertEquals("The current user should be returned.", "admin", data.user.name);
+    }
+};
+
+gpii.express.couchuser.test.server.caseHolder.verifyLoggedOut = function (response, body) {
+    gpii.express.couchuser.test.server.caseHolder.isSaneResponse(response, body);
+
+    var data = typeof body === "string" ? JSON.parse(body) : body;
+    jqUnit.assertTrue("The response should be 'ok'.", data.ok);
+    jqUnit.assertNotNull("There should be a user returned.", data.user);
+    if (data.user) {
+        jqUnit.assertEquals("The current user should be returned.", "admin", data.user.name);
+    }
+};
+
+gpii.express.couchuser.test.server.caseHolder.verifyCurrentUserLoggedOut = function (response, body) {
+    gpii.express.couchuser.test.server.caseHolder.isSaneResponse(response, body, 401);
+
+    var data = typeof body === "string" ? JSON.parse(body) : body;
+    jqUnit.assertFalse("The response should not be 'ok'.", data.ok);
+    jqUnit.assertUndefined("There should not be a user returned.", data.user);
+};
+
+
+gpii.express.couchuser.test.server.caseHolder.verifyFailedLogin = function (response, body) {
+    gpii.express.couchuser.test.server.caseHolder.isSaneResponse(response, body, 500);
+    var data = typeof body === "string" ? JSON.parse(body) : body;
+    jqUnit.assertFalse("The response should not be 'ok'.", data.ok);
+    jqUnit.assertUndefined("There should not be a user returned.", data.user);
+};
+
+gpii.express.couchuser.test.server.caseHolder.verifyUnverifiedLogin = function (response, body) {
+    gpii.express.couchuser.test.server.caseHolder.isSaneResponse(response, body, 401);
+    var data = typeof body === "string" ? JSON.parse(body) : body;
+    jqUnit.assertFalse("The response should not be 'ok'.", data.ok);
+    jqUnit.assertUndefined("There should not be a user returned.", data.user);
+};
+
+gpii.express.couchuser.test.server.caseHolder.verifyDuplicateEmailBlocked = function (response, body) {
```

Comment:
All of these "verify" functions are essentially identical. Consolidate into a single function which accepts more arguments (such as "statusCode", "absentFields" (for "user") etc.)